// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package secdata_transfer;

//  
service SecdataHandle {
  // Sends a greeting
  rpc TransferKlineData (KlineRequest) returns (KlineReply) {}
  
  rpc AskKlineData (AskKlineReq) returns (AskKlineRsp) {}
}

//  
message KlineRequest {
  int32 ts = 1;
  string code = 2;
  string period = 3;
  double open = 4;
  double high = 5;
  double low = 6;
  double close = 7;
  int64 vol = 8;
  double amount = 9;
}

//  
message KlineReply {
  int32 status = 1;
  string message = 2;
}

//  
message AskKlineReq {
  string code = 1;
  string period = 2;
  int32 from_ts = 3;
  int32 to_ts = 4;
}

message Kline{
  int32 ts = 1;
  double open = 2;
  double high = 3;
  double low = 4;
  double close = 5;
  int64 vol = 6;
  double amount = 7;
}

message AskKlineRsp {
  int32 status = 1;
  string message = 2;
  repeated Kline klines=3;
}

